great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -196,15 +144,20 @@ def selectable(self):
     def use_quoted_name(self):
         return self._use_quoted_name
 
-    def _create_temporary_table(  # noqa: C901, PLR0912
-        self, temp_table_name, query, temp_table_schema_name=None
-    ) -> str:
+    def _create_temporary_table(  # noqa: C901, PLR0912, PLR0915
+        self, dialect, query, temp_table_schema_name=None
+    ) -> Tuple[str, str]:
         """
         Create Temporary table based on sql query. This will be used as a basis for executing expectations.
         :param query:
         """
 
-        dialect: GXSqlDialect = self.dialect
+        temp_table_name = generate_temporary_table_name()
+        # mssql expects all temporary table names to have a prefix '#'
+        if dialect == GXSqlDialect.MSSQL:
+            temp_table_name = f"#{temp_table_name}"
+
+        dialect = self.dialect
```

Comment:
this was logic that was shared, so it was brought into the `_create_temporary_table()` method.